js/mainjs/config.js

```diff
@@ -33,7 +33,10 @@ function configManager(path) {
 	 */
 	config.attr = function(key, value) {
 		if (value !== undefined) {
-			this[key] = value;
+			config[key] = value;
+		}
+		if (config[key] === undefined) {
+			config[key] = null;
```

Comment:
I don't understand this pattern. Is this fixing a problem that you found, or is it just defensive programming? (there's nothing wrong with defensive programming, I'm just trying to understand what the change is doing)